scripts/update_envoy.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Copyright 2020 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Update the Envoy SHA in istio/proxy WORKSPACE with the first argument (aka ENVOY_SHA) and
+# the second argument (aka ENVOY_SHA commit date)
+
+# Exit immediately for non zero status
+set -e
+# Check unset variables
+set -u
+# Print commands
+set -x
+
+UPDATE_BRANCH=${UPDATE_BRANCH:-"master"}
```

Comment:
Are we going to update for every envoy commit? That is ~10x a day, which will also lead to 10x updates to istio/istio per day. I am concerned about the load/churn this will have

Reply:
oh, maybe update per day set as periodic job?

Reply:
No this should be a periodic job. I think every week is a good start.

Reply:
every week? ok, let me make it triggered on a weekly basis.

Reply:
It could be every commit for the release branches maybe since I think we manually pull those in (I think - verify with Peter or others though). Agree on periodic for the master branch.

Thanks!

Reply:
Release branch already does commit on per commit base, so yeah that has been working fine.